core/src/main/scala/org/apache/spark/util/ThreadUtils.scala

```diff
@@ -156,4 +156,11 @@ private[spark] object ThreadUtils {
         result
     }
   }
+
+  /**
+   * Construct a new ForkJoinPool with a specified max parallelism.
+   */
+  def newForkJoinPool(maxThreadNumber: Int): scala.concurrent.forkjoin.ForkJoinPool = {
```

Comment:
I figured it would be best to put it here with the rest of the ThreadUtils stuff. If we add the custom factory for the name as @zsxwing asked gets complicated enough probably worth keeping here in a utility function.